### ISSUE TITLE
dynamic providers / components in pulumi yaml

### ISSUE THREAD

**Issue by terekete** [pulumi/pulumi-yaml#308]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
To be able to generate a yaml definition for a pulumi resource based on a custom component or dynamic provider.

### Affected area/feature
Flexibility to utilize custom components in yaml.

**Comment by Frassle** [comment]
I think yaml should already work with multi-language components, same as all our other languages.
Like the other languages you can't just call into another language library without it being setup to be called into, i.e. packaged https://www.pulumi.com/docs/guides/pulumi-packages/

**Comment by AaronFriel** [comment]
@terekete We have an example using Pulumi's EKS provider, which uses custom components written in TypeScript: https://github.com/pulumi/pulumi-yaml/blob/main/examples/aws-eks/Pulumi.yaml

Is that the sort of thing you're referring to or expand on what provider(s) you're trying to use?

**Comment by terekete** [comment]
Thanks! Did not see this as an option, interesting. So if I read this right. given I can package my component, that can be used by Pulumi YAML, is that correct ?

**Comment by AaronFriel** [comment]
Yes, I think that's right. If code generation works for a component provider, that is, it has a valid schema.json, then YAML should be able to use that too. The `GetSchema` API call must be implemented so that Pulumi YAML can validate the resources & functions created.
